### PR TITLE
feat: add ability to inject external logger and pass AWS credentials through input props

### PR DESCRIPTION
**Pull request by rdziarnowski**
Adds ability to inject external logger (i.e. github actions core logger) and allow externally injected credentials to AWS instead of just env variables.

### CODE REVIEW

**Review by mattlewis92 — COMMENTED** [review]
This LGTM, but will leave it to @Justinon to give the final approval as he wrote this library

File: src/helpers/Interfaces.ts [hunk 1]
```diff
@@ -87,7 +88,7 @@ export interface IDeployToGroupProps {
   /**
    * Every level below the specified log level is silenced. Defaults to INFO.
    */
-  readonly logLevel?: log.LogLevelDesc;
+  readonly logLevel?: LogLevel;
```
Comment (Justinon) [inline src/helpers/Interfaces.ts:91]: Why are we redefining types that are already existent as an enum in `log.LogLevelDesc`?

File: src/deploy.ts [hunk 2]
```diff
@@ -126,19 +129,33 @@ async function deployAppVersionsToGroup(client: ElasticBeanstalkClient, props: I
   if (triggerErr.errors.length !== 0) throw triggerErr;
 }
 
+/**
+ * Get AWS Credentials if provided
+ */
+function getCredentials(props: IDeployToGroupProps) {
```
Comment (Justinon) [inline src/deploy.ts:135]: I'm not sure I understand the motivation for doing this?

Comment (rdziarnowski) [reply src/deploy.ts:135]: to have ability to pass credentials as arguments as well. This is needed for js type of github actions (reusable ones) as you cannot pass env to them.

File: src/helpers/Interfaces.ts (hunk 1 above)
Comment (rdziarnowski) [reply src/helpers/Interfaces.ts:91]: To make them generic, so that loglevel doesn't have to be explicitly added as dependency on upper layer.

**Review by Justinon — APPROVED** [review]

**Pull request merged**